bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestBot.java

```diff
@@ -165,11 +167,43 @@ public List<WorkItem> getPeriodicItems() {
         if (processPR) {
             List<PullRequest> prs = poller.updatedPullRequests();
             workItems.addAll(getPullRequestWorkItems(prs));
+
+            var jCheckConfUpdateRelatedPRs = getJCheckConfUpdateRelatedPRs();
+            // Filter out duplicate prs
+            var filteredPrs = jCheckConfUpdateRelatedPRs.stream()
+                    .filter(pullRequest -> prs.stream()
+                            .noneMatch(pr -> pr.isSame(pullRequest)))
+                    .toList();
+            workItems.addAll(getPullRequestWorkItems(filteredPrs));
             poller.lastBatchHandled();
         }
         return workItems;
     }
 
+    private List<PullRequest> getJCheckConfUpdateRelatedPRs() {
+        var ret = new ArrayList<PullRequest>();
+        // Get all branches except 'pr' branch
+        var allTargetRefs = remoteRepo.branches().stream()
+                .map(HostedBranch::name)
+                .filter(name -> !name.startsWith("pr/"))
+                .toList();
```

Comment:
Currently, fetching all existing branches is the main overhead in this feature. It takes about 2s for JDK repo. If this overhead is not acceptable, the other way to solve this is store the branches in bot config.

Reply:
Yes, it will be much better. No extra overhead will be added.